src/storage/core.rs

```diff
@@ -1016,12 +1022,18 @@ where
 
     /// Core deletion logic, when lock on `Safe<K>` is acquired
     async fn delete_core(&self, safe: &Safe<K>, key: &K, only_if_presented: bool) -> Result<u64> {
-        let deleted_in_active = Self::mark_all_as_deleted_active(safe, key, only_if_presented).await?;
+        let deleted_in_active_result = Self::mark_all_as_deleted_active(safe, key, only_if_presented).await?;
+        let deleted_in_active = deleted_in_active_result.as_ref().map(|r| if r.success() { 1 } else { 0 }).unwrap_or(0);
         let deleted_in_closed = Self::mark_all_as_deleted_closed(safe, key).await?;
 
         if deleted_in_closed > 0 {
             self.observer.defer_dump_old_blob_indexes().await;
         }
+        if let Some(result) = deleted_in_active_result {
+            if self.inner.should_try_fsync(result.dirty_bytes()) {
```

Comment:
I think this can be done inside `mark_all_as_deleted_active`. In this way the code changes will be within the single function

Reply:
We don't have inner or observer inside `mark_all` function, so it can't be moved here without adding 2 more args to the function call